### PR TITLE
Tutorial 31 - Introduction to material science

### PR DESCRIPTION
**Pull request by arunppsg**
This PR is a tutorial on using DeepChem for material science. It introduces materials featurizers, datasets and demonstrate one example. Happy to incorporate any feedback! In particular, wondering if some context should be added or removed.

### CODE REVIEW

File: examples/tutorials/31_Introduction_To_Material_Science.ipynb [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "d43c4bc3",
+   "metadata": {},
+   "source": [
+    "## Featurizers <a class=\"anchor\" id=\"featurizers\"></a>\n",
+    "\n",
+    "### Material Structure Featurizers <a class=\"anchor\" id=\"crystal-featurizers\"></a>\n",
+    "\n",
+    "Crystal are gemoetric structures which has to be featurized for using in machine learning algorithms.  The following featurizers provided by DeepChem helps in featurizing crystals:\n",
```
Comment (rbharath) [inline examples/tutorials/31_Introduction_To_Material_Science.ipynb:127]: Typo: geometric

File: examples/tutorials/31_Introduction_To_Material_Science.ipynb [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+    "\n",
+    "- The [SineCoulombMatrix](https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html#sinecoulombmatrix) featurizer a crystal by calculating sine coulomb matrix for the crystals. It can be called using `dc.featurizers.SineCoulombMatrix` function.\n",
+    "- The [CGCNNFeaturizer](https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html#cgcnnfeaturizer) calculates structure graph features of crystals. It can be called using `dc.featurizers.CGCNNFeaturizer` function. For more details on CGCNNFeaturizer, refer [1].\n",
+    "- The [LCNNFeaturizer](https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html#lcnnfeaturizer) calculates the 2-D Surface graph features in 6 different permutations. It can be used using the utility `dc.feat.LCNNFeaturizer`.\n",
+    "\n",
+    "[1] T. Xie and J. C. Grossman, “Crystal graph convolutional neural networks for an accurate and interpretable prediction of material properties”, Phys. Rev. Lett. 120, 2018, https://arxiv.org/abs/1710.10324"
```
Comment (rbharath) [inline examples/tutorials/31_Introduction_To_Material_Science.ipynb:133]: Can you add a citation for lattice convolutions to this paper: https://pubs.acs.org/doi/10.1021/acs.jpcc.9b03370

**Review by rbharath — COMMENTED** [review]
This is excellent @arunppsg! Really good work :). I have a couple of minor comments below but this is looking pretty solid to me already.

@ncfrey Could you do a quick review as well?

File: examples/tutorials/31_Introduction_To_Material_Science.ipynb [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+   "id": "d43c4bc3",
+   "metadata": {},
+   "source": [
+    "## Featurizers <a class=\"anchor\" id=\"featurizers\"></a>\n",
+    "\n",
+    "### Material Structure Featurizers <a class=\"anchor\" id=\"crystal-featurizers\"></a>\n",
+    "\n",
+    "Crystal are gemoetric structures which has to be featurized for using in machine learning algorithms.  The following featurizers provided by DeepChem helps in featurizing crystals:\n",
+    "\n",
+    "- The [SineCoulombMatrix](https://deepchem.readthedocs.io/en/latest/api_reference/featurizers.html#sinecoulombmatrix) featurizer a crystal by calculating sine coulomb matrix for the crystals. It can be called using `dc.featurizers.SineCoulombMatrix` function.\n",
```
Comment (ncfrey) [inline examples/tutorials/31_Introduction_To_Material_Science.ipynb:129]: You can add the reference for sine coulomb matrix here as well

File: examples/tutorials/31_Introduction_To_Material_Science.ipynb [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+    "print(\"Training set score:\", model.evaluate(train_dataset, [metric], transformers))\n",
+    "print(\"Test set score:\", model.evaluate(test_dataset, [metric], transformers))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df2e9207",
+   "metadata": {},
+   "source": [
+    "# Congratulations! Time to join the Community!\n",
```
Comment (ncfrey) [inline examples/tutorials/31_Introduction_To_Material_Science.ipynb:395]: If you want to add links for further reading here, feel free to grab some from my list or blog post, or link to them directly: https://github.com/ncfrey/resources
https://towardsdatascience.com/getting-started-in-materials-informatics-41ee34d5ccfe

**Review by ncfrey — COMMENTED** [review]
This looks great! 

Since this tutorial doesn't exactly follow the sequence of the other tutorials, I'm wondering if it makes sense to break it out into a separate directory like `materials_tutorials` or `application_tutorials` that would organize tutorials outside the "core" deepchem series.
What do you think @rbharath ?

**Comment by arunppsg** [comment]
Thanks for feedback @ncfrey @rbharath. Made the fixes regarding the same.

**Pull request merged**